lib/codecs/src/decoding/format/syslog.rs

```diff
@@ -47,8 +48,8 @@ impl SyslogDeserializerConfig {
             .optional_field("msgid", Kind::bytes(), None)
             .optional_field("procid", Kind::integer().or_bytes(), None)
             // "structured data" in a syslog message can be stored in any field, but will always be
-            // a string.
-            .unknown_fields(Kind::bytes())
+            // a map of string -> string.
+            .unknown_fields(Kind::object(BTreeMap::default()))
```

Comment:
cc/ @JeanMertz does this schema adjustment make sense?

Reply:
Unfortunately, no.

This is a trap I've run into many times before.

`Kind::object(BTreeMap::default())` means "An object without any unknown fields, and an empty map of known fields". Which translates to "an object that is known to be empty.

What you want instead is: `Kind::object(Collection::any())`

Reply:
Actually, if it's marked as `a map of string -> string` as the comment suggests, we can improve on this one:

`Kind::object(Collection::from_unknown(Kind::bytes()))`, translating to "an object with no known fields, and one or more unknown fields of the bytes type".

Reply:
Ah! Gotcha, yeah I see it now. That makes sense.